Increase test coverage: Array.prototype.toString

matedabis opened the pull request:
The coverage is measured by the following script, using `--jerry-test-suite --test262 --unittests --jerry-tests`:
https://github.com/matedabis/jerryscript/blob/gcov_coverage_tester/tests/gcov-tests/gcovtester.py

Branch coverage:
Before: 5/6
After:  6/6

JerryScript-DCO-1.0-Signed-off-by: Mate Dabis mdabis@inf.u-szeged.hu

rerobika commented on tests/jerry/array-prototype-tostring.js:
```diff
@@ -61,3 +61,10 @@ try {
+
+// Checking behavior when undefined is passed to the function
+try {
+  Array.prototype.toString.call(undefined);
```
Add an `assert(false)` after this line.

rerobika commented on tests/jerry/array-prototype-tostring.js:
```diff
@@ -61,3 +61,10 @@ try {
   assert (e instanceof ReferenceError);
 }
+
+// Checking behavior when undefined is passed to the function
```
At this point not an `undefined` passed to the function. The function's `this_argument` is `undefined`.

matedabis commented:
I updated the patch according to the review.

matedabis commented:
Updated the patch:
- Added comment for the case containing which part of the ES5.1 it tests.
- Ran the test code in Gecko, V8, SpiderMonkey engines and got the same result.

rerobika commented on tests/jerry/array-prototype-tostring.js:
```diff
@@ -61,3 +61,12 @@ try {
 }
+
+/* ES 5.1 15.4.4.2 1. */
+// Checking behavior when the function's this_argument is undefined
```
If the related part of the standard is present, than use the following format:

```
/* ES v5.1, 15.4.4.2.1. 
   Checking behavior when the function's this_argument is undefined */
```

matedabis commented:
Changed comment style.

matedabis commented:
Closing PR because of moving to another branch.
